### PR TITLE
Add Windows Min Artifacts Building on Jenkins 

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add Windows Min Artifacts Building on Jenkins 

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/2306

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/system/temporary_directory.py [hunk 1]
```diff
@@ -17,9 +16,21 @@
...
-        func(path)
+    logging.debug(f"excvalue {excvalue}")
+    logging.debug(f"func {func.__name__}")
+    if func in (os.rmdir, os.remove, os.unlink):
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO | stat.S_IWRITE | stat.S_IREAD)  # 0777 nix* / +wr windows
+        try_total = 3
+        for try_count in range(try_total):  # Re-run func to force deletion especially on windows
+            try:
+                logging.debug(f'Try count: {try_count + 1}/{try_total}')
+                func_result = None
```
Comment (dblock) [inline src/system/temporary_directory.py:27]: Unnecessary assign here.

File: src/system/temporary_directory.py [hunk 2]
```diff
@@ -17,9 +16,21 @@
...
 def g__handleRemoveReadonly(func: FunctionType, path: str, exc: Any) -> Any:
     excvalue = exc[1]
-    if func in (os.rmdir, os.remove, os.unlink) and excvalue.errno == errno.EACCES:
-        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)  # 0777
-        func(path)
+    logging.debug(f"excvalue {excvalue}")
+    logging.debug(f"func {func.__name__}")
+    if func in (os.rmdir, os.remove, os.unlink):
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO | stat.S_IWRITE | stat.S_IREAD)  # 0777 nix* / +wr windows
+        try_total = 3
```
Comment (dblock) [inline src/system/temporary_directory.py:23]: Rename to `retry_count`

File: src/system/temporary_directory.py [hunk 3]
```diff
@@ -17,9 +16,21 @@
...
+        for try_count in range(try_total):  # Re-run func to force deletion especially on windows
```
Comment (dblock) [inline src/system/temporary_directory.py:24]: Rename to `retry`

File: src/system/temporary_directory.py [hunk 4]
```diff
@@ -17,9 +16,21 @@
...
+            try:
+                logging.debug(f'Try count: {try_count + 1}/{try_total}')
+                func_result = None
+                func_result = func(path)
+                if func_result is None:
+                    break
+            except Exception as ex:
+                logging.debug(func_result)
```
Comment (dblock) [inline src/system/temporary_directory.py:32]: Do `logging.warn` of the actual error inside `ex`.

I would remove `logging.debug`'s.

**Review by dblock — COMMENTED** [review]
This is fine. 

I think we should raise the bar on the retry implementation and take a separate function from https://stackoverflow.com/questions/42521549/retry-function-in-python for example.

**Comment by peterzhuamazon** [comment]
Hi @dblock I agree with many points you raised but not agree with putting the retries into another function.
The `func(path)` is specifically to `onerror` and requires a specific `path` param.
If we want to make a function that is generalized for everything we need to write complicated code just to take `x` number of params from different functions which is not very suitable.

File: src/system/temporary_directory.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/system/temporary_directory.py:27]: removed.

File: src/system/temporary_directory.py (hunk 2 above)
Comment (peterzhuamazon) [reply src/system/temporary_directory.py:23]: renamed.

File: src/system/temporary_directory.py (hunk 3 above)
Comment (peterzhuamazon) [reply src/system/temporary_directory.py:24]: renamed

File: src/system/temporary_directory.py (hunk 4 above)
Comment (peterzhuamazon) [reply src/system/temporary_directory.py:32]: changed to warn.

**Review by zelinh — APPROVED** [review]

**Pull request merged**